modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SSource.java

```diff
@@ -216,6 +220,19 @@ public void write() {
         visitor.visit(WriterConstants.CLASS_VERSION, classAccess, className, null, classBase, classInterfaces);
         visitor.visitSource(Location.computeSourceName(name, source), null);
 
+        // Write the a method to bootstrap def calls
```

Comment:
I do it this way because `def` is always going to need `getRuntimeClass` (or something like it) to be specific for each generated class. So each generated class needs its own static reference to the appropriate `Definition`. Lifting the method to `PainlessScript` would require resolving the `Definition` with reflection or a method reference or something.

I could also play around with making the bootstrap method non-static. If I did that I could indeed move this to `PainlessScript`. I didn't think that'd work when I was working on it the first time but now I think it might. Do you think I should have a look at that?